src/webgl/interaction.js

```diff
@@ -106,4 +114,369 @@ p5.prototype.orbitControl = function(sensitivityX, sensitivityY) {
+          'post',
+          this._grid.call(
+            this,
+            arguments[1],
+            arguments[2],
+            arguments[3],
+            arguments[4],
+            arguments[5]
+          )
+        );
+      } else if (arguments[0] === constants.AXES) {
+        this.registerMethod(
+          'post',
+          this._axesIcon.call(
+            this,
+            arguments[1],
+            arguments[2],
+            arguments[3],
+            arguments[4]
+          )
+        );
+      }
+    } else {
+      if (arguments[0] === constants.GRID) {
+        this.registerMethod('post', this._grid.call(this));
+      } else if (arguments[0] === constants.AXES) {
+        this.registerMethod('post', this._axesIcon.call(this));
+      } else if (arguments[0] === constants.OFF) {
+        return;
+      }
+    }
+  } else {
+    this.registerMethod('post', this._grid.call(this));
+    this.registerMethod('post', this._axesIcon.call(this));
+  }
+};
+
+/**
+ * For use with debugMode
+ * @private
+ * @method _grid
+ * @param {Number} [size] size of grid sides
+ * @param {Number} [div] number of grid divisions
+ * @param {Number} [xOff] offset of grid center from origin in X axis
+ * @param {Number} [yOff] offset of grid center from origin in Y axis
+ * @param {Number} [zOff] offset of grid center from origin in Z axis
+ */
+p5.prototype._grid = function(size, numDivs, xOff, yOff, zOff) {
+  if (typeof size === 'undefined') {
+    size = this.width / 2;
+  }
+  if (typeof numDivs === 'undefined') {
+    // ensure at least 2 divisions
+    numDivs = Math.round(size / 30) < 4 ? 4 : Math.round(size / 30);
+  }
+  if (typeof xOff === 'undefined') {
+    xOff = 0;
+  }
+  if (typeof yOff === 'undefined') {
+    yOff = 0;
+  }
+  if (typeof zOff === 'undefined') {
+    zOff = 0;
+  }
+
+  var spacing = size / numDivs;
+  var halfSize = size / 2;
+
+  return function() {
+    this.push();
+
+    this._renderer.uMVMatrix.set(
+      this._renderer._curCamera.cameraMatrix.mat4[0],
+      this._renderer._curCamera.cameraMatrix.mat4[1],
+      this._renderer._curCamera.cameraMatrix.mat4[2],
+      this._renderer._curCamera.cameraMatrix.mat4[3],
+      this._renderer._curCamera.cameraMatrix.mat4[4],
+      this._renderer._curCamera.cameraMatrix.mat4[5],
+      this._renderer._curCamera.cameraMatrix.mat4[6],
+      this._renderer._curCamera.cameraMatrix.mat4[7],
+      this._renderer._curCamera.cameraMatrix.mat4[8],
+      this._renderer._curCamera.cameraMatrix.mat4[9],
+      this._renderer._curCamera.cameraMatrix.mat4[10],
+      this._renderer._curCamera.cameraMatrix.mat4[11],
+      this._renderer._curCamera.cameraMatrix.mat4[12],
+      this._renderer._curCamera.cameraMatrix.mat4[13],
+      this._renderer._curCamera.cameraMatrix.mat4[14],
+      this._renderer._curCamera.cameraMatrix.mat4[15]
+    );
+
+    // Lines along X axis
+    for (var q = 0; q <= numDivs; q++) {
+      this.strokeWeight(1);
+      this.stroke(0, 0, 0);
```

Comment:
Good point, that would be nice!  

I've set it to use the last-set stroke color and stroke weight.  This introduces an odd behavior when called without first setting a stroke, where it will use the stroke color last set by `_axesIcon()`, but I suspect this is fixed by #3076.